test/framework/management/kind/mgmt.go

```diff
@@ -49,33 +55,25 @@ type Cluster struct {
 
 // NewCluster sets up a new kind cluster to be used as the management cluster.
 func NewCluster(ctx context.Context, name string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
-	cmd := exec.NewCommand(
-		exec.WithCommand("kind"),
-		exec.WithArgs("create", "cluster", "--name", name),
-	)
-	return create(ctx, cmd, name, scheme, images...)
+	p := cluster.NewProvider(cluster.ProviderWithLogger(cmd.NewLogger()))
+	return create(ctx, p, name, nil, scheme, images...)
 }
 
 // NewClusterWithConfig creates a kind cluster using a kind-config file.
 func NewClusterWithConfig(ctx context.Context, name, configFile string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
-	cmd := exec.NewCommand(
-		exec.WithCommand("kind"),
-		exec.WithArgs("create", "cluster", "--name", name, "--config", configFile),
-	)
-	return create(ctx, cmd, name, scheme, images...)
+	p := cluster.NewProvider(cluster.ProviderWithLogger(cmd.NewLogger()))
+	config := cluster.CreateWithConfigFile(configFile)
+	return create(ctx, p, name, config, scheme, images...)
 }
 
-func create(ctx context.Context, cmd *exec.Command, name string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
+func create(ctx context.Context, provider *cluster.Provider, name string, config cluster.CreateOption, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
 	f, err := ioutil.TempFile("", "mgmt-kubeconfig")
```

Comment:
Doesn't look like this is being used anywhere except to set the `KubeconfigPath` of the `Cluster` struct, if we aren't telling the provider to use the temp file anymore, should this be removed?

Reply:
This is getting used to specify the kubeconfig path later on, i think the comment here was made before those changes were made though. this is resolved.